autoload/lsp.vim

```diff
@@ -866,6 +879,26 @@ function! lsp#get_whitelisted_servers(...) abort
     return call(function('lsp#get_allowed_servers'), a:000)
 endfunction
 
+function! s:get_allowed_servers_vim(buffer_filetype) abort
```

Comment:
I'm wondering if we can move all `get_allowed_servers` logic to internal folder in a complete different folder.

```
autload/lsp/internal/get_allowed_servers.vim
```

`lsp.vim` is getting quite big so makes it difficult to refactor.

we should still have a top level `lsp#get_allowed_servers(s:servers, filetype)` but will internally call the other file.